geopandas/geodataframe.py

```diff
@@ -653,6 +719,13 @@ def __finalize__(self, other, method=None, **kwargs):
         else:
             for name in self._metadata:
                 object.__setattr__(self, name, getattr(other, name, None))
+
+            # find GeometryArrays and propagate CRS to their self counterparts
+            for col, dtype in other.dtypes.iteritems():
+                if isinstance(dtype, GeometryDtype):
+                    if col in self.columns:
+                        self[col].crs = other[col].crs
```

Comment:
Did you have a case where this is needed? 
As normally, this finalize should handle dataframe-level metadata, not necessarily array-level metadata (I would assume the arrays should preserve their crs information. Although this is the first time I am dealing with array-level metadata myself, not certainly not sure)